docs: add GitLab DORA metrics guide (#2324)
port-labs/port-docs: docs/guides/all/set-up-deployments-dora-gitlab.md

---
sidebar_position: 11
displayed_sidebar: null
---

import Tabs from "@theme/Tabs"
import TabItem from "@theme/TabItem"

# Set up deployments using GitLab merge requests or jobs

> This guide adapts the [Create & track DORA metrics in your portal](/guides/all/create-and-track-dora-metrics-in-your-portal) guide by demonstrating how to define deployments in Port using GitLab's Merge Requests (MRs) and Jobs.
> By the end of this guide, you’ll be able to leverage DORA metrics using GitLab and gain accurate deployment insights in your portal. 

Reviewer: ```suggestion
The [Create & track DORA metrics in your portal](/guides/all/create-and-track-dora-metrics-in-your-portal) guide walks you through the installation of the DORA metrics experience in Port, using GitHub as the default Git provider.

This guide assumes you have implemented the aforementioned guide, and shows how to adapt it for GitLab users. 

It will show how to define deployments in Port using GitLab's Merge Requests (MRs) and Jobs.  
By the end of this guide, you will be able to leverage DORA metrics using GitLab and gain accurate deployment insights in your portal. 
```

## Prerequisites

To set up GitLab merge requests or jobs for DORA metrics in your portal, you will need:

- Admin permissions (in order to install the experience and execute self-service actions).
- An installed DORA Metrics experience - based on this guide: [Create & track DORA metrics in your portal](/guides/all/create-and-track-dora-metrics-in-your-portal)

> - A connected GitLab integration.

Reviewer: ```suggestion
- An installed GitLab integration.
```

## Configure your deployments

1. **Map merge requests and/or jobs as deployments**
    - Go to the **Builder** -> **Data Sources** then click on the GitLab integration
    - Add a new mapping to ingest GitLab merge requests as deployment entities.

> :::note

Reviewer: use info instead of note, and add a title

> Keep in mind that the mapping ingests the deployments into the dora_deployment_event blueprint, for further information visit Create & track DORA metrics in your portal.

Reviewer: ```suggestion
Keep in mind that the mapping ingests the deployments into the `dora_deployment_event` blueprint, for further information visit Create & track DORA metrics in your portal.
```
